### PR TITLE
Refactor/tue status

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Still missing:
As a target would specify a branch, that should be ignored as well. We could either a file in the root of the git directory or store it somewhere else. But this file should also be removed, If the version is removed from the target.

I am not sure if this should already be implemented right now or could be done later.

### CODE REVIEW

File: setup/tue-functions.bash [hunk 1]
```diff
@@ -64,16 +64,22 @@ function tue-apt-select-mirror
 #                                       GIT LOCAL HOUSEKEEPING
 # ----------------------------------------------------------------------------------------------------
 
-function __tue-git-default-branch
+function _tue-git-get-default-branch
+{
+    # Takes current dir in case $1 is empty
+    git -C "$1" symbolic-ref refs/remotes/origin/HEAD | sed 's@^refs/remotes/origin/@@'
```
Comment (ar13pit) [inline setup/tue-functions.bash:70]: Add a check: If this command fails, then use `git remote show origin | grep HEAD | awk '{print $3}'`

Creation of symbolic-ref is not very robust. I have seen it to be absent many times.

File: setup/tue-functions.bash [hunk 2]
```diff
@@ -485,12 +496,28 @@ function _tue-repo-status
 
             local current_branch
             current_branch=$(git -C "$pkg_dir" rev-parse --abbrev-ref HEAD)
-            case $current_branch in
-                master|develop|indigo-devel|hydro-devel|jade-devel|kinetic-devel|toolchain-2.9) ;;
-                *) _robocup_branch_allowed "$current_branch" || echo -e "\033[1m$name\033[0m is on branch '$current_branch'";;
-            esac
-        fi
 
+            local test_branches="master"
```
Comment (ar13pit) [inline setup/tue-functions.bash:500]: Initialization to master is not needed.

File: setup/tue-functions.bash [hunk 3]
```diff
@@ -485,12 +496,28 @@ function _tue-repo-status
...
+                    [ -z "$status" ] && return 0
+                    # else
+                    allowed="true"
+                    break
+                fi
+            done
+            [ "$allowed" != "true" ] && echo -e "\033[1m$name\033[0m is on branch '$current_branch'"
+        fi
         vctype=git
     elif [ -d "$pkg_dir"/.svn ]
```
Comment (ar13pit) [inline setup/tue-functions.bash:522]: Still want to keep this? None of our repos use svn

**Comment by ar13pit** [comment]
Also I did not understand what you are trying to say in the missing part.

File: setup/tue-functions.bash (hunk 1 above)
Comment (MatthijsBurgh) [reply setup/tue-functions.bash:70]: Based on being empty? Or by return code?

File: setup/tue-functions.bash (hunk 3 above)
Comment (MatthijsBurgh) [reply setup/tue-functions.bash:522]: I don't see any reason to remove it right now, also not in scope of this PR.

File: setup/tue-functions.bash (hunk 1 above)
Comment (ar13pit) [reply setup/tue-functions.bash:70]: Error code. When symbolic-ref fails, the error printed out is
```
fatal: ref refs/remotes/origin/HEAD is not a symbolic ref
```

File: setup/tue-functions.bash (hunk 3 above)
Comment (ar13pit) [reply setup/tue-functions.bash:522]: Okay.

**Comment by MatthijsBurgh** [comment]
> Also I did not understand what you are trying to say in the missing part.

For example, `kdl_parser`, the default branch of the repo is `melodic-devel`. But if you are on Noetic, the target specifies `noetic-devel`. But this will keep showing up in tue-status. This is not desired as tue-status should be empty in case nothing is modified.
So we should also allow branches defined in the target for that repo, also in case that branch isn't the default branch.

File: setup/tue-functions.bash (hunk 2 above)
Comment (MatthijsBurgh) [reply setup/tue-functions.bash]: Fixed

**Comment by ar13pit** [comment]
> > Also I did not understand what you are trying to say in the missing part.
> 
> For example, `kdl_parser`, the default branch of the repo is `melodic-devel`. But if you are on Noetic, the target specifies `noetic-devel`. But this will keep showing up in tue-status. This is not desired as tue-status should be empty in case nothing is modified.
> So we should also allow branches defined in the target for that repo, also in case that branch isn't the default branch.

I see. I think we should create a file in the workspace (.env directory) where repos with such a setting are listed out. tue-status checks this file and allows that branch.

**Comment by ar13pit** [comment]
This looks good conceptually. But I haven't tested it. Have you tested it thoroughly? I can approve it then.

**Comment by MatthijsBurgh** [comment]
I have tested it significantly, but it can wait till tomorrow.

**Pull request merged**